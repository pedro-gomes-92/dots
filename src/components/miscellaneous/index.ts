export * from './Image';
export * from './Icon';
export * from './Close';
export * from './Helper';
export * from './Overlay';
export * from './Divider';
export * from './NavBar';
export * from './Pagination';
