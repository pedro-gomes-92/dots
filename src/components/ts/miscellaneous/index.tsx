export * from './Image';
export * from './Icon';
export * from './Helper';
export * from './Overlay';
export * from './Divider';
export * from './NavBar';
export * from './Hero';
export * from './Pagination';
