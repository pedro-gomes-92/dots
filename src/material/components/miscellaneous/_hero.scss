/*
 * hero.scss
 *
 * All styling applied to the hero component
 *
 * Index
 * - Hero
 *    - Color
 */

.hero {
  height: 100%;

  &.is {
    @each $name, $pair in $colors {
      // Color
      &-#{$name} {
        background-color: nth($pair, 1);
        color: nth($pair, 2);

        &-inverted {
          background-color: nth($pair, 2);
          color: nth($pair, 1);
        }
      }
    }
  }
}
