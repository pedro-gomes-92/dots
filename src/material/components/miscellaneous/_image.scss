/*
 * image.scss
 *
 * All styling applied to the image component
 *
 * Index
 * - Image
 *    - Size
 */

.image {
  &.is {
    // Size
    &-small {
      width: $image-size-small;
    }

    &-normal {
      width: $image-size-normal;
    }

    &-medium {
      width: $image-size-medium;
    }

    &-large {
      width: $image-size-large;
    }
  }
}
