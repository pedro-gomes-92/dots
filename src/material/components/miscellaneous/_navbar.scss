/*
 * navbar.scss
 *
 * All styling applied to the navbar component
 *
 * Index
 * - NavBar
 * - NavBar Menu
 *    - Trigger
 *    - Opened
 * - NavBar Brand
 */

.navbar {
  margin: 0;
  height: $navbar-height;
  min-height: auto;
  padding: $navbar-padding;
  border: none;

  > .container {
    min-height: 100%;
  }
}

.navbar-menu {
  &-trigger {
    @include screen($navbar-device-inline-active) {
      display: none;
    }
  }

  &.is {
    // Opened
    &-opened {
      @extend .is-active;

      .navbar-item > .button {
        width: 100%;
      }

      @include screen($navbar-device-inline-active) {
        .navbar-item > .button {
          width: auto;
        }
      }
    }
  }
}

.navbar-brand {
  justify-content: space-between;
}
