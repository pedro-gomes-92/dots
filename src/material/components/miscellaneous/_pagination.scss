/*
 * pagination.scss
 *
 * All styling applied to the pagination component
 *
 * Index
 * - Pagination
 *    - List
 *    - Item
 */

.pagination {
  justify-content: center;

  // List
  &-list {
    border: none;

    @include flex-auto;

    > .list-item {
      padding: 0;
    }
  }

  // Item
  &-link {
    &.is-current {
      font-weight: bold;
      @include hover {
        color: $pagination-current-color;
        border-color: $pagination-hover-border-color;
      }

      @include active {
        color: $pagination-current-color;
        border-color: $pagination-focus-border-color;
      }
    }
  }
}
