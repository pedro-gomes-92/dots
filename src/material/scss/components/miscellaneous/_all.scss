@import 'icon';
@import 'image';
@import 'overlay';
@import 'divider';
@import 'hero';
@import 'navbar';
@import 'pagination';
