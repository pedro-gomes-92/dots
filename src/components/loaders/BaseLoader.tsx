import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../Base';
import { IconProps } from '../miscellaneous/Icon';

export interface BaseLoaderProps extends BaseProps {}

export interface BaseLoaderTypeProps extends BaseLoaderProps {
  sizeIcon?: IconProps['size'];
  icon?: IconProps['name'];
}

export const BaseLoader: FunctionComponent<BaseLoaderProps> = ({
  className,
  children,
  ...rest
}: BaseLoaderProps): JSX.Element => {
  return (
    <Base className={classnames('loaders', className)} {...rest}>
      {children}
    </Base>
  );
};
