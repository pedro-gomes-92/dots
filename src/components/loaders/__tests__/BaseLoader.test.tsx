import React from 'react';
import { render } from '@testing-library/react';

import { BaseLoader } from '../BaseLoader';
import { Icon } from '../../miscellaneous';

describe('Given <BaseLoader />', (): void => {
  describe('When initializes', (): void => {
    let container: HTMLElement;
    it('Then should match the snapshot', (): void => {
      container = render(<BaseLoader />).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses loader', (): void => {
      beforeEach((): void => {
        container = render(<BaseLoader loader={<Icon name="menu" />} />).container;
      });
      it('Then should match the snapshot', (): void => {
        expect(container).toMatchSnapshot();
      });

      it('And should render loader', (): void => {
        expect(container.querySelector('.icon-font-menu')).toBeTruthy();
      });
    });
  });
});
