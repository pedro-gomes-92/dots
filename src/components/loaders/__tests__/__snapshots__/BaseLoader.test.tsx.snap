// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <BaseLoader /> When initializes And uses loader Then should match the snapshot 1`] = `
<div>
  <div
    class="loaders"
  >
    <span
      class="icon"
    >
      <i
        class="icon-font icon-font-menu"
      />
    </span>
  </div>
</div>
`;

exports[`Given <BaseLoader /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="loaders"
  >
    <span
      class="icon is-loader is-relative"
    >
      <i
        class="icon-font icon-font-loader"
      />
    </span>
  </div>
</div>
`;
